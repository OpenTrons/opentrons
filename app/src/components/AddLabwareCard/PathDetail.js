// @flow

import * as React from 'react'

import { InputField } from '@opentrons/components'
import styles from './styles.css'

// TODO(mc, 2019-11-18): i18n
const CUSTOM_LABWARE_DEFINITIONS_FOLDER = 'Custom Labware Definitions Folder'

export type PathDetailProps = {|
  path: string,
|}

export function PathDetail(props: PathDetailProps) {
  return (
    <div className={styles.path_detail}>
      <h4 className={styles.path_detail_title}>
        {CUSTOM_LABWARE_DEFINITIONS_FOLDER}
      </h4>
      <InputField
        readOnly
        value={props.path}
        type="text"
        onFocus={(e: SyntheticFocusEvent<HTMLInputElement>) => {
          e.currentTarget.select()
        }}
      />
    </div>
  )
}
