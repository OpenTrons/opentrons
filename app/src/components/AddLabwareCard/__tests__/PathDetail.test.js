// @flow
import * as React from 'react'
import { shallow } from 'enzyme'

import { PathDetail } from '../PathDetail'

describe('AddLabwareCard', () => {
  const mockPath = '/path/to/a/place'

  test('component displays path', () => {
    const wrapper = shallow(<PathDetail path={mockPath} />)

    expect(wrapper.html()).toContain(mockPath)
  })

  test('component renders', () => {
    const wrapper = shallow(<PathDetail path={mockPath} />)

    expect(wrapper).toMatchSnapshot()
  })
})
