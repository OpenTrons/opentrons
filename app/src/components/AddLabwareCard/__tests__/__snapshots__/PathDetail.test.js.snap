// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLabwareCard component renders 1`] = `
<div
  className="path_detail"
>
  <h4
    className="path_detail_title"
  >
    Custom Labware Definitions Folder
  </h4>
  <InputField
    onFocus={[Function]}
    readOnly={true}
    type="text"
    value="/path/to/a/place"
  />
</div>
`;
